### PR TITLE
reduce graph code

### PR DESCRIPTION
**Pull request by star-e**
Remove unused graph vertex/edge remove.
Remove unused graph property map.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

The pull request simplifies graph-related code by removing unused methods and properties, reducing complexity and potential maintenance overhead.

- **`cocos/rendering/custom/layout-graph.ts`**: Removed methods related to vertex and edge management.
- **`cocos/rendering/custom/render-graph.ts`**: Removed comments and sections related to the 'PropertyGraph Concept'.
- **`cocos/rendering/custom/layout-graph-utils.ts`**: No significant changes noted.
- **`native/cocos/renderer/pipeline/custom/details/GraphImpl.h`**: No significant changes noted.

Ensure thorough testing to validate that these removals do not affect any dependent functionality or introduce regressions.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/rendering/custom/layout-graph.ts [hunk 1]
```diff
@@ -115,27 +115,6 @@ export class LayoutGraphVertex {
 
 //-----------------------------------------------------------------
 // PropertyGraph Concept
```
Comment (minggo) [inline cocos/rendering/custom/layout-graph.ts]: Should also remove the comment?

File: cocos/rendering/custom/layout-graph.ts [hunk 2]
```diff
@@ -265,120 +239,19 @@ export class LayoutGraph implements BidirectionalGraph
...
     //-----------------------------------------------------------------
     // NamedGraph
     vertexName (v: number): string {
         return this._names[v];
     }
-    vertexNameMap (): LayoutGraphNameMap {
-        return new LayoutGraphNameMap(this._names);
-    }
     //-----------------------------------------------------------------
     // PropertyGraph
```
Comment (minggo) [inline cocos/rendering/custom/layout-graph.ts]: Remove comment?

**Review by minggo — APPROVED** [review]

File: cocos/rendering/custom/layout-graph.ts (hunk 2 above)
Comment (star-e) [reply cocos/rendering/custom/layout-graph.ts]: Removed.

File: cocos/rendering/custom/layout-graph.ts (hunk 1 above)
Comment (star-e) [reply cocos/rendering/custom/layout-graph.ts]: Removed.

**Review by minggo — APPROVED** [review]

**Pull request merged**